.github/workflows/merge.yml

```diff
@@ -0,0 +1,55 @@
+name: Merge
+
+on:
+  push:
+    branches: [ "main" ]
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.0
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Build with Gradle
+        run: chmod +x ./gradlew && ./gradlew build
```

Comment:
As far as my understanding goes build still invokes all unit tests across all modules/core/app elements of the projects.

As far as I can see it doesn't execute the Android/UI tests. Does it make sense to have a separate action that executes the Android/UI tests?
I suspect it's the same setup but just `./gradlew connectedAndroidTest`? Does it make sense to add this as a step after the first one right in this job or should it be a separate job? (I am leaning more to a separate job to parallelize this, on the other hand it might not be needed for the current scale ...).

Reply:
> As far as my understanding goes build still invokes all unit tests across all modules/core/app elements of the projects.

Yes, that's correct 👍 

I pointed it out in our last Android meeting that we are not executing UI tests, and I would include it next in separate PR. Reason is that it does not simply involve running `./gradlew connectedAndroidTest`, but it requires additional setup beforehand, such e.g. setting up an emulator where the ui tests will run. Haven't ever done it with Github actions before, hence would like to address it separately, because I suspect I might need some commits to get it running correctly 😅

Reply:
Sounds good! Let's tackle this in a separate PR 👍

Reply:
Created this issue https://github.com/StanfordSpezi/SpeziKt/issues/21 and suggested in addition to implement a screen recording mechanism 👍